### DISCUSSION TITLE
Multiple BottomSheetModalProvider

### DISCUSSION THREAD

**Discussion by Zack-Heisnberg** [gorhom/react-native-bottom-sheet#1273] [Ideas]
Hello , 

did you know that you can use Multiple BottomSheetModalProvider , 

For example in an app i am working on right now , 

we needed the bottonsheets to be above the Navigation for most of the case , 

but we also needed a single bottomsheet to be Over the navigation

i could achieve this by using a BottomSheetModalProvider on my
```
 <BottomSheetModalProvider>
      <Stack.Navigator
      </Stack.Navigator>
    </BottomSheetModalProvider>
```

and inside that screen only , even the screen is nested under the first BottomSheetModalProvider  , 
the context get changed and the BottomSheetModal  would look for only the first BottomSheetModalProvider  it's finds

```
 <BaseLayout>
      <BottomSheetModalProvider>
        <Header toggleColorScheme={toggleColorScheme} colorScheme={colorScheme} />
          <BottomSheetModal ..... /> // Navigation here would be on top of the bottomsheet
        </SafeAreaView>
      </BottomSheetModalProvider>
    </BaseLayout>
```

do you think my approach is correct ? , and should that be in docs ?

**Comment by ShooterArk** [comment]
@Zack-Heisnberg thanks alot for sharing.
